src/testing-library/render.ts

```diff
@@ -0,0 +1,86 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import type { StoryFnMubanReturnType } from '@muban/storybook/dist/client/preview/types';
+import type { Story } from '@muban/storybook/dist/client/preview/types-6-0';
+import { createApp } from '@muban/test-mock';
+import type { BaseStoryFn } from '@storybook/addons/dist/ts3.9/types';
+import { getQueriesForElement, prettyDOM, queries } from '@testing-library/dom';
+import type { PrettyDOMOptions } from '@testing-library/dom/types/pretty-dom';
+import * as refQueries from './queryByRef';
+
+type StoryArgs<T extends Story> = T extends BaseStoryFn<infer R, StoryFnMubanReturnType>
+  ? R
+  : Record<string, any>;
+
+const mountedWrappers = new Set<HTMLDivElement>();
+
+/**
+ * Render a Muban Story into `document.body`, so tests can be run after.
+ * @param componentFactory A Muban Story (function that returns a component definition object).
+ * @param [templateData] An optional object with data that is used by the template to render the
+ * initial HTML markup.
+ *
+ * @returns { container, debug, unmount, html, ...queries} An object of useful information and helpers.
+ *
+ * - `container` HTML Element that the component is rendered in.
+ * - `debug` A function that outputs a pretty version of the component HTML.
+ * - `unmount` Removes the container from the HTML so the component will be unmounted
+ * - `html` Returns the raw outerHTML
+ * - `...queries` All element queries bound to the component container
+ */
+export function render<T extends Story>(componentFactory: T, templateData?: StoryArgs<T>) {
+  if (typeof componentFactory !== 'function') {
+    throw new TypeError('Story must be a function');
+  }
+  const { component, appComponents, template } = (componentFactory as any)();
```

Comment:
It seems that TypeScript doesn't properly narrow down the type to `StoryFn` here, but we can cast it to `() => StoryFnMubanReturnType` to at least retain somewhat accurate types instead of just `any`.